runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -6436,7 +6437,7 @@ TR::Register *J9::Power::TreeEvaluator::VMnewEvaluator(TR::Node *node, TR::CodeG
 
       TR::Instruction *firstInstruction = cg->getAppendInstruction();
 
-      if (!isDualTLH && needZeroInit)
+      if ((!isDualTLH || disableClearBatch) && needZeroInit)
          {
```

Comment:
all redundant code to me.  looking at needZeroInit, your `disableClearBatch`  should be equivalent to `!fej9->tlhHasBeenCleared()`.  i am wondering if `!fej9->tlhHasBeenCleared()` returns wrong value, otherwise your code is just redundant.  please look into that code to make sure.

Reply:
ultimately, whether setting `TR_DisableBatchClear`  is effective or not (i.e. the meaning) is reflected through `fej9->tlhHasBeenCleared()`.

Reply:
`fej9->tlhHasBeenCleared()` returns correct value. However, the logic `!isDualTLH && needZeroInit` is not correct. This is because, it assumes that if DualTLH is used (which is true), then there is no need to do explicit zeroInit. I have removed the redundant logic.